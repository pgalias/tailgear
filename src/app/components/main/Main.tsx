import React, { FunctionComponent } from 'react';
import { Route, Switch } from 'react-router-dom';
import { RoutesList } from '../../routes';
import styles from './main.module.css';

export const Main: FunctionComponent = () => (
  <main className={styles.main}>
    <div className={styles.wrapper}>
      <Switch>
        <Route path="/" exact>
          Honey, I`m home
        </Route>
        <Route path="/contribution">Coming soon</Route>
        <RoutesList />
      </Switch>
    </div>
  </main>
);
