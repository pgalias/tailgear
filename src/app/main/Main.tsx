import React, { FunctionComponent } from 'react';
import { Route, Switch } from 'react-router-dom';
import { RoutesList } from '../components';
import styles from './main.module.css';

export const Main: FunctionComponent = () => (
  <main className={styles.main}>
    <Switch>
      <Route path="/" exact>
        Honey, I`m home
      </Route>
      {RoutesList()}
    </Switch>
  </main>
);
