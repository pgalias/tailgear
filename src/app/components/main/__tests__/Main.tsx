import React from 'react';
import { render, screen } from '@testing-library/react';
import { Main } from '../Main';

describe('app::components::Main', () => {
  test('should render properly', () => {
    render(
      <Main>
        <span>child</span>
      </Main>
    );

    expect(screen.getByRole('main')).toMatchSnapshot();
  });
});
