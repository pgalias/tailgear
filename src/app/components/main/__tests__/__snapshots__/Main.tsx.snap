// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app::components::Main should render properly 1`] = `
<main
  class="main"
>
  <div
    class="wrapper"
  >
    <span>
      child
    </span>
  </div>
</main>
`;
